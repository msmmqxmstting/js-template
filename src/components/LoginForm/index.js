import { Button, Form, Input, Row } from 'antd';
import { HXIcon } from 'components/Common/StatelessComponent';
import { inject, observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

const FormItem = Form.Item;

@withRouter
@inject('appStore')
@observer
class LoginForm extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}
	formRef = React.createRef();
	static propTypes = {
		onLoginCallback: PropTypes.func
	};

	static defaultProps = {
		onLoginCallback: () => {}
	};

	onFinish = async (values) => {
		const { loginSubmit, loading } = this.props.appStore;

		values.type = 'HEXIN';

		if (loading) {
			return;
		}
		const loginRes = await loginSubmit(values);
		if (loginRes && loginRes.code === 200) {
			this.props.history.push('/');
		}
	};

	render() {
		const { loading } = this.props.appStore;

		const inputStyle = { height: '50px', fontSize: '16px' };
		const createInputLabel = (text) => (
			<div className={'label-row'}>
				<span className={'label-text'}>{text}</span>
			</div>
		);
		const LoginFormEle = (
			<Form ref={this.formRef} onFinish={this.onFinish}>
				{createInputLabel('登录账号')}
				<FormItem
					style={{ zIndex: 1 }}
					name={'username'}
					rules={[{ required: true, message: '请输入用户名！' }]}
				>
					<Input
						style={inputStyle}
						prefix={<HXIcon className={'hx-prefix-icon'} type="hx-user" />}
						placeholder={'请输入您的用户名'}
					/>
				</FormItem>
				{createInputLabel('登录密码')}
				<FormItem
					style={{ height: '35px', zIndex: 1 }}
					name={'password'}
					rules={[{ required: true, message: '请输入密码！' }]}
				>
					<Input.Password
						style={inputStyle}
						prefix={<HXIcon className={'hx-prefix-icon'} type="hx-lock" />}
						placeholder="请输入您的密码"
					/>
				</FormItem>
				<div id="login-btn-wrap">
					<Button
						type="primary"
						htmlType="submit"
						className="login-form-btn block-btn"
						loading={loading}
						disabled={loading}
					>
						立即登录
					</Button>
				</div>
			</Form>
		);

		return (
			<div>
				<Row style={{ height: '100%' }}>
					<div className={'screen-title-wrap'}>
						<p>和信保理业务平台</p>
					</div>
					<div style={{ textAlign: 'center' }}>
						<div id="form-input-wrap">{LoginFormEle}</div>
					</div>
				</Row>
			</div>
		);
	}
}
export default LoginForm;
