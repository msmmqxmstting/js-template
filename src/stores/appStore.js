import { STORAGE_KEY_MAPPING } from 'constants/user';
import { action, observable } from 'mobx';

import { login, logout } from 'services/app';
import { openNotification, showSuccessMsg } from 'utils/antd';
import { getLocalUserInfo, saveUserInfo } from 'utils/storage';

class AppStore {
	@observable
	userInfo;
	@observable
	isLogined;
	@observable
	loading;
	@observable
	downloadProgress;

	constructor() {
		this.userInfo = getLocalUserInfo();
		this.isLogined = window.localStorage.getItem(STORAGE_KEY_MAPPING.IS_LOGINED);
		this.loading = false;
		this.downloadProgress = 0;
	}

	@action.bound
	loginSubmit = async (values) => {
		this.showLoading();
		const loginRes = await login(values);
		this.hideLoading();
		if (loginRes && loginRes.code !== 200) {
			openNotification('error', '登录失败', loginRes.message);
			return;
		}
		this.userInfo = loginRes;
		this.isLogined = true;
		saveUserInfo(loginRes);
		showSuccessMsg('登录成功');
		return loginRes;
	};

	@action.bound
	logout(successcb = () => {}) {
		logout().then((res) => {
			window.localStorage.clear();
			showSuccessMsg('退出成功', 1);
			this.isLogined = false;
			successcb();
		});
	}

	//全局加载
	@action
	showLoading() {
		this.loading = true;
	}

	@action
	hideLoading() {
		this.loading = false;
	}
	//设置下载的进度条，一般针对文件的请求
	@action
	setDownloadProgress(progress = 0) {
		let complete = progress;
		if (progress < 0) {
			complete = 0;
		}
		if (complete > 100) {
			complete = 100;
		}
		this.downloadProgress = complete;
	}
}

const appStore = new AppStore();

export default appStore;
export { AppStore };
