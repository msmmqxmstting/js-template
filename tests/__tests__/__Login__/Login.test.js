/**
 * Created By xun  on 2018-11-20 08:54.
 * Description: __Login__.test
 */

import React from 'react';
import LoginPage from 'pages/Login';
import LoginForm from 'components/LoginForm';
import { appStore } from 'stores';
import { MOCK_REACT_ROUTER, mountWrap, shallowWrap } from '../../tools/contextWrap';
import { Router } from 'react-router-dom';
import * as stores from 'stores';
import { Provider } from 'mobx-react';

const loginFormMount = () =>
	mountWrap(
		<Router history={MOCK_REACT_ROUTER.history}>
			<LoginForm appStore={appStore} />
		</Router>
	);
describe('登录页面', () => {
	/*	beforeAll(() => {
		console.log('test1 before all');
	});

	afterAll(() => {
		console.log('test1 after all');
	});

	beforeEach(() => {
		console.log('test1 before each');
	});

	afterEach(() => {
		console.log('test1 after each');
	})
	*/
	it('页面正确渲染快照', () => {
		const wrapper = shallowWrap(
			<Router history={MOCK_REACT_ROUTER.history}>
				<Provider {...stores}>
					<LoginPage appStore={appStore} />
				</Provider>
			</Router>
		);
		expect(wrapper.html()).toMatchSnapshot();
	});
	it('表单正确渲染', () => {
		//如果目标组件包含 withRouter属性，必须用<Router>组件包裹
		const wrapper = loginFormMount();
		const btn = wrapper.find('#login-btn-wrap');
		expect(btn.text()).toBe('立即登录');
	});
	it('输入点击发送登录请求', () => {
		const wrapper = loginFormMount();

		wrapper.find('input#username').simulate('change', { target: { value: '111' } });
		wrapper.find('input#password').simulate('change', { target: { value: '222' } });

		expect(wrapper.find('input#username').prop('value')).toBe('111');
		expect(wrapper.find('input#password').prop('value')).toBe('222');
		//组件被装饰器包裹，是一个高阶组件，需要拿instance之前做下find操作，这样才能拿到真实组件的实例
		const instance = wrapper.find('LoginForm').instance();
		// 模拟点击事件
		wrapper.find('#login-btn-wrap').at(0).simulate('click');
		const spyFn = jest.spyOn(instance, 'onFinish');
		instance.onFinish();

		expect(spyFn).toHaveBeenCalled();
		spyFn.mockRestore();
	});
});
