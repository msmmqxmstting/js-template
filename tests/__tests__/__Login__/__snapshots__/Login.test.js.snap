// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`登录页面 页面正确渲染快照 1`] = `"<div><div id=\\"login-wrap\\"><div id=\\"logo-wrap\\"><img src=\\"[object Object]\\" alt=\\"logo\\"/></div><div id=\\"form-wrap\\"><div><div class=\\"ant-row\\" style=\\"height:100%\\"><div class=\\"screen-title-wrap\\"><p>和信保理业务平台</p></div><div style=\\"text-align:center\\"><div id=\\"form-input-wrap\\"><form class=\\"ant-form ant-form-horizontal\\"><div class=\\"label-row\\"><span class=\\"label-text\\">登录账号</span></div><div class=\\"ant-row ant-form-item\\" style=\\"z-index:1\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper\\" style=\\"height:50px;font-size:16px\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" class=\\"anticon hx-prefix-icon\\"><svg width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"\\"><use xlink:href=\\"#hx-user\\"></use></svg></span></span><input type=\\"text\\" placeholder=\\"请输入您的用户名\\" value=\\"\\" id=\\"username\\" class=\\"ant-input\\"/></span></div></div></div></div><div class=\\"label-row\\"><span class=\\"label-text\\">登录密码</span></div><div class=\\"ant-row ant-form-item\\" style=\\"height:35px;z-index:1\\"><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-password\\" style=\\"height:50px;font-size:16px\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" class=\\"anticon hx-prefix-icon\\"><svg width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"\\"><use xlink:href=\\"#hx-lock\\"></use></svg></span></span><input type=\\"password\\" placeholder=\\"请输入您的密码\\" action=\\"click\\" value=\\"\\" id=\\"password\\" class=\\"ant-input\\"/><span class=\\"ant-input-suffix\\"><span role=\\"img\\" aria-label=\\"eye-invisible\\" tabindex=\\"-1\\" class=\\"anticon anticon-eye-invisible ant-input-password-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"eye-invisible\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z\\"></path><path d=\\"M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z\\"></path></svg></span></span></span></div></div></div></div><div id=\\"login-btn-wrap\\"><button type=\\"submit\\" class=\\"ant-btn ant-btn-primary login-form-btn block-btn\\"><span>立即登录</span></button></div></form></div></div></div></div></div></div></div>"`;
